homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,332 @@
+    SensorEntityDescription(
+        key=TYPE_UV,
+        translation_key="uv_index",
+        native_unit_of_measurement="Index",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WEEKLYRAININ,
+        translation_key="weekly_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=2,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDDIR,
+        translation_key="wind_direction",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDGUSTMPH,
+        translation_key="wind_gust",
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.WIND_SPEED,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDSPEEDMPH,
+        translation_key="wind_speed",
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.WIND_SPEED,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_YEARLYRAININ,
+        translation_key="yearly_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        suggested_display_precision=2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Ambient Network sensor platform."""
+
+    coordinator: AmbientNetworkDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is not None:
+        async_add_entities(
+            AmbientNetworkSensor(
+                coordinator,
+                description,
+                coordinator.config_entry.data[ENTITY_STATION_NAME],
+            )
+            for description in SENSOR_DESCRIPTIONS
+            if AmbientNetworkSensor.get_sensor_value(coordinator.data, description.key)
+            is not None
+        )
+
+
+class AmbientNetworkSensor(AmbientNetworkEntity, SensorEntity):
+    """A sensor implementation for an Ambient Weather Network sensor."""
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: SensorEntityDescription,
+        station_name: str,
+    ) -> None:
+        """Initialize a sensor object."""
+
+        super().__init__(coordinator, description, station_name)
+        self._attr_suggested_display_precision = description.suggested_display_precision
```

Comment:
Isn't this already done by setting an entity description to the entity?

Reply:
Yes. Fixed.